engine/src/available-command.ts

```diff
@@ -337,22 +342,32 @@ export function possibleSpecialActions(engine: Engine, player: Player) {
   return commands;
 }
 
-export function possibleBoardActions(engine: Engine, player: Player) {
-  const commands = [];
+export function possibleBoardActions(actions: BoardActions, p: PlayerObject): AvailableCommand[] {
+  const commands: AvailableCommand[] = [];
+
+  // not allowed if everything is lost - see https://github.com/boardgamers/gaia-project/issues/76
+  const canGain = (reward: Reward) => {
+    const type = reward.type;
+    const limit = resourceLimits[type];
+    return limit == null || p.data.getResources(type) < limit;
+  };
 
   let poweracts = BoardAction.values(Expansion.All).filter(
     (pwract) =>
-      engine.boardActions[pwract] === null && engine.player(player).canPay(Reward.parse(boardActions[pwract].cost))
+      actions[pwract] === null &&
+      p.data.canPay(Reward.parse(boardActions[pwract].cost)) &&
+      canGain(Reward.parse(boardActions[pwract].income[0])[0])
```

Comment:
```suggestion
      boardActions[pwract].income.some(income => Reward.parse(income).some(reward => canCain(reward)))
```

Just in case we add board actions in the future, or a mod changes how current board actions work